pkg/hintrunner/zero/zerohint_memcpy_test.go

```diff
@@ -21,6 +22,116 @@ func TestZeroHintMemcpy(t *testing.T) {
+				operanders: []*hintOperander{
+					{Name: "arrayPtr", Kind: apRelative, Value: addr(5)},
+					{Name: "elmSize", Kind: fpRelative, Value: feltInt64(4)},
+					{Name: "nElms", Kind: fpRelative, Value: feltInt64(0)},
+					{Name: "index", Kind: uninitialized},
+					{Name: "key", Kind: uninitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSearchSortedLowerHint(
+						ctx.operanders["arrayPtr"],
+						ctx.operanders["elmSize"],
+						ctx.operanders["nElms"],
+						ctx.operanders["key"],
+						ctx.operanders["index"],
+					)
+				},
+				errCheck: errorTextContains("Invalid value for n_elms"),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "arrayPtr", Kind: apRelative, Value: addr(0)},
+					{Name: "elmSize", Kind: fpRelative, Value: feltInt64(4)},
+					{Name: "nElms", Kind: fpRelative, Value: feltInt64(4)},
+					{Name: "index", Kind: fpRelative, Value: feltInt64(2)},
+					{Name: "key", Kind: fpRelative, Value: feltInt64(2)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSearchSortedLowerHint(
+						ctx.operanders["arrayPtr"],
+						ctx.operanders["elmSize"],
+						ctx.operanders["nElms"],
+						ctx.operanders["key"],
+						ctx.operanders["index"],
+					)
+				},
+				errCheck: errorTextContains("failed obtaining the variable: __find_element_max_size"),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "array.el0", Kind: apRelative, Value: feltInt64(0)},
+					{Name: "array.el1", Kind: apRelative, Value: feltInt64(1)},
+					{Name: "array.el2", Kind: apRelative, Value: feltInt64(2)},
+					{Name: "array.el3", Kind: apRelative, Value: feltInt64(3)},
+					{Name: "arrayPtr", Kind: apRelative, Value: addr(0)},
+					{Name: "elmSize", Kind: fpRelative, Value: feltInt64(4)},
+					{Name: "nElms", Kind: fpRelative, Value: feltInt64(8)},
+					{Name: "index", Kind: fpRelative, Value: feltInt64(2)},
+					{Name: "key", Kind: fpRelative, Value: feltInt64(2)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSearchSortedLowerHint(
+						ctx.operanders["arrayPtr"],
+						ctx.operanders["elmSize"],
+						ctx.operanders["nElms"],
+						ctx.operanders["key"],
+						ctx.operanders["index"],
+					)
+				},
+				errCheck: errorTextContains("find_element() can only be used with n_elms<="),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "array.el0", Kind: apRelative, Value: feltInt64(0)},
+					{Name: "array.el1", Kind: apRelative, Value: feltInt64(1)},
+					{Name: "array.el2", Kind: apRelative, Value: feltInt64(2)},
+					{Name: "array.el3", Kind: apRelative, Value: feltInt64(3)},
+					{Name: "arrayPtr", Kind: apRelative, Value: addr(0)},
+					{Name: "elmSize", Kind: fpRelative, Value: feltInt64(4)},
+					{Name: "nElms", Kind: fpRelative, Value: feltInt64(9)},
+					{Name: "index", Kind: fpRelative, Value: feltInt64(3)},
+					{Name: "key", Kind: fpRelative, Value: feltInt64(0)},
```

Comment:
I believe these values don't make a lot of sense.
- `nElms` should be the amount of elements in the array
- `index` should have been `uninitialized` kind, because this is something that will be assigned during the execution of the hint. In fact, instead of checking you get a nil error after running the test, you should check the value that was saved in `index` address. This variable will represent the index of the first element of the array that is bigger or equal to the value of `key`
- `key` should have the value you are looking for. You can play with this in the tests, for example if you have array [0, 1, 2, 3] as in this case, apart from trying simple cases, you can try something edge case like `key = 5`. Since all elements of the array are smaller then the else statement after the for should be executed getting `index = 4`.

A suggestion is to try an array with different numbers than the ones you are using so the values of the elements don't coincide with their actual indexes. Notice the array has to be sorted in increasing order